src/driver/types.rs

```diff
@@ -43,7 +43,7 @@ impl HeadInfo {
         };
 
         let tx_calldata = first_tx.input.clone();
```

Comment:
Are we double cloning here since we clone the input, then immediately call `to_vec` on it which I believe will clone under the hood? Same thing for `try_from_ecotone_block`.

Reply:
Oops, good catch, just fixed!